### PR TITLE
PHP8 support

### PR DESCRIPTION
**Pull request by jlouism**

### CODE REVIEW

**Comment by KmeCnin** [comment]
## Summary of changes

### 1) PHP 8 / 8.2 compatibility hardening
- Replaced deprecated `sizeOf()` calls with `count()` across multiple files (commit message mentions **32 occurrences across 10 files**).
- Added explicit `public` visibility to methods that previously relied on implicit visibility (**237 methods across 22 class files**, per commit message).
- Fixed PHP 8+ incompatibility in `implode()` usage:  
  changed from `implode($array, ',')` to `implode(',', $array)` in  
  `scheduledBackup::validateRotateMethod()` (the old argument order breaks on PHP 8+).
- Fixed additional PHP 8.1 / 8.2 issues mentioned in commit notes:
  - missing spaces in `public function` declarations,
  - `dbConnection::query()` signature aligned with its `mysqli` parent,
  - nullable parameter deprecation fix in `MergeException`.

### 2) Error handling modernization
- Removed the error suppression operator `@` in three places (e.g. `@chmod`, `@fopen`), relying instead on explicit checks and error handling.

### 3) Test suite added (new coverage)
- Added a PHP 8.2 validation test runner: `tests/php82-validation.php` (commit message: **39 tests**), covering:
  - minimum PHP version check (>= 8.2),
  - syntax validation of PHP class files,
  - class loading and validation methods,
  - detection/reporting of deprecated functions (`sizeOf`, `ereg*`, etc.).
- Added broader automated tests:
  - **Integration tests** (commit message: **37 tests**) using an in-memory SQLite database.
  - **Unit tests** (commit message: **105 tests**) focusing on validation methods and edge cases.
- Added documentation in `tests/README.md`.

**Comment by jlouism** [comment]
> ## Summary of changes
> ### 1) PHP 8 / 8.2 compatibility hardening
> 
>     * Replaced deprecated `sizeOf()` calls with `count()` across multiple files (commit message mentions **32 occurrences across 10 files**).
> 
>     * Added explicit `public` visibility to methods that previously relied on implicit visibility (**237 methods across 22 class files**, per commit message).
> 
>     * Fixed PHP 8+ incompatibility in `implode()` usage:
>       changed from `implode($array, ',')` to `implode(',', $array)` in
>       `scheduledBackup::validateRotateMethod()` (the old argument order breaks on PHP 8+).
> 
>     * Fixed additional PHP 8.1 / 8.2 issues mentioned in commit notes:
>       
>       * missing spaces in `public function` declarations,
>       * `dbConnection::query()` signature aligned with its `mysqli` parent,
>       * nullable parameter deprecation fix in `MergeException`.
> 
> 
> ### 2) Error handling modernization
> 
>     * Removed the error suppression operator `@` in three places (e.g. `@chmod`, `@fopen`), relying instead on explicit checks and error handling.
> 
> 
> ### 3) Test suite added (new coverage)
> 
>     * Added a PHP 8.2 validation test runner: `tests/php82-validation.php` (commit message: **39 tests**), covering:
>       
>       * minimum PHP version check (>= 8.2),
>       * syntax validation of PHP class files,
>       * class loading and validation methods,
>       * detection/reporting of deprecated functions (`sizeOf`, `ereg*`, etc.).
> 
>     * Added broader automated tests:
>       
>       * **Integration tests** (commit message: **37 tests**) using an in-memory SQLite database.
>       * **Unit tests** (commit message: **105 tests**) focusing on validation methods and edge cases.
> 
>     * Added documentation in `tests/README.md`.

C'est a peu pres ca

**Review by KmeCnin — APPROVED** [review]

**Review by MaelitoP — APPROVED** [review]

File: includes/backupRestorer.class.php [hunk 1]
```diff
@@ -70,7 +70,7 @@ function restoreLocal($backupSnapshot, $path) {
 					throw new Exception('backupRestorer->restoreLocal: '."Error: Attempted to create directory for restore, but failed -- $path");
 				}
 				// Set permissions to be rwx owner and rx group.
-				if( ! @chmod($path, 0750) ) {
+				if( ! chmod($path, 0750) ) {
 					throw new Exception('backupRestorer->restoreLocal: '."Error: Attempted to change permissions for newly created restore path, but failed -- $path");
```
Comment (arnaud-lb) [inline includes/backupRestorer.class.php]: Here too
